diff --git a/tools/releasetools/build_image.py b/tools/releasetools/build_image.py
index 334f6601c..17ec747f5 100755
--- a/tools/releasetools/build_image.py
+++ b/tools/releasetools/build_image.py
@@ -193,6 +193,10 @@ def MakeVerityEnabledImage(out_file, prop_dict):
                              block_dev, signer_path, signer_key):
     shutil.rmtree(tempdir_name, ignore_errors=True)
     return False
+    
+  print "backup system image to some place for re-sign with dm keys"
+   # build the full verified image
+  shutil.copy2(out_file , "/home/dhcui/system-kaios-sprd.img")
 
   # build the full verified image
   if not BuildVerifiedImage(out_file,
